Google.Api.Generator.Tests/WhitespaceFormatterTest.cs

```diff
@@ -0,0 +1,150 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This test loads this source file during the test, parses it into
+// Roslyn, removes all trivia, passes it through the Whitespace formatter,
+// then checks that the input source and re-formatted source are identical.
+// Code at the start, end, and between IGNORE_START/IGNORE_END markers
+// are ignored.
+// So some of the source code looks a bit odd, as it needs to test every
+// code construct is formatted correctly.
+// TODO: Add further code as WhitespaceFormatter adds more formatting.
+
+// IGNORE_END
+using Google.Api.Generator.Formatting;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Xunit;
+using sys = System;
+
+namespace Google.Api.Generator.Tests
+{
+    public class WhitespaceFormatterTest
+    {
+        // Test nested class; base-list with multiple items.
+        // Test aliased type.
+        private class Inner : List<sys::String>, IEnumerable<string>, IList<string>
+        {
+            // Test ctor with no parameters.
+            // Test ctor calling another ctor with multiple parameters.
+            // Test empty bock body.
+            public Inner() : this(0, 1, 2)
+            {
+            }
+
+            // Test ctor with single parameter.
+            // Test ctor calling ctor with no args.
+            // Test expression body.
+            public Inner(int a) : this() => _a = a + 1;
+
+            // Test ctor with multiple parameters.
+            // Test non-empty block body.
+            public Inner(int a, int b, int c)
+            {
+                // Test assignment.
+                _a = a;
+                // Test binary operator.
+                _b = b + c;
+            }
+
+            // Test instance field with an initializer.
+            private readonly int _a = -1;
+
+            // Test instance field with no intializer.
+            private readonly int _b;
+
+            // Test get-only auto-property.
+            public string P1 { get; }
+
+            // Test get/set auto-property with initializer.
+            public string P2 { get; set; } = "Hello_world!";
+
+            // Test get-only expression-bodied property.
+            public string P3
+            {
+                get => "Hello_another_world!";
+            }
+
+            private string _p4 = "";
+
+            // Test get/set expression-bodied property.
+            public string P4
+            {
+                get => _p4 + _p4;
+                set => _p4 = value;
+            }
+        }
+
+        // Test base-list with one item.
+        private class EmptyInner : List<string>
+        {
+            // Test class that has no members.
+        }
+        // IGNORE_START
```

Comment:
Now we get the start... but I don't know what it means (yet).

Reply:
ack, see earlier response.